src/app/core/contentful.service.ts

```diff
@@ -32,8 +32,12 @@ export class ContentfulService {
 
   // get event metadata from backend
   async getEventMetadata(name?: string): Promise<any> {
-    const event = await this.http.get(`${environment.apiUrl}/event`,
+    const event: any = await this.http.get(`${environment.apiUrl}/event`,
       name && { params: new HttpParams().set('name', name) }).toPromise();
+    // Only public events are requestable on production mode
+    if (environment.production && !event.isPublic) {
```

Comment:
Should there also be a check if the event is defined or not, to get a better error for that? Or does Contentful always respond with some event?

Reply:
In dev mode, it gives 500 Internal server error when trying to access event that is not there. At least on quick test on dev mode server.

Reply:
If event for given name is not found, that `http.get` throws 500. If we would like to handle that error better we would need to wrap the request in try-catch. Right now this if-block is reached only when event has been received.

Reply:
If Contentful responds with error 500 for a name that doesn't exist, I guess then there shouldn't be a case for event to be undefined. With the current use-case for this, I guess it's okay for now.

Reply:
The error propably comes from backend, and not from Contentful itself